src/Build/Evaluation/Evaluator.cs

```diff
@@ -1651,15 +1652,17 @@ private List<ProjectRootElement> ExpandAndLoadImports(string directoryOfImportin
 
                 string extensionPathExpanded = _data.ExpandString(extensionPath);
 
-                if (!_fallbackSearchPathsCache.DirectoryExists(extensionPathExpanded))
+                var newExpandedCondition = importElement.Condition.Replace(extensionPropertyRefAsString, extensionPathExpanded, StringComparison.OrdinalIgnoreCase);
+                if (!EvaluateConditionCollectingConditionedProperties(importElement, newExpandedCondition, ExpanderOptions.ExpandProperties, ParserOptions.AllowProperties,
+                            _projectRootElementCache))
                 {
                     continue;
                 }
 
-                var newExpandedCondition = importElement.Condition.Replace(extensionPropertyRefAsString, extensionPathExpanded, StringComparison.OrdinalIgnoreCase);
-                if (!EvaluateConditionCollectingConditionedProperties(importElement, newExpandedCondition, ExpanderOptions.ExpandProperties, ParserOptions.AllowProperties,
-                            _projectRootElementCache))
+                if (!_fallbackSearchPathsCache.DirectoryExists(extensionPathExpanded))
```

Comment:
I have had to read this and the old version . . . too many times . . . to get to this understanding so I think a comment here would be helpful:

```suggestion
                // If the whole fallback folder doesn't exist, short-circuit and don't
                // bother constructing an exact file path.
                if (!_fallbackSearchPathsCache.DirectoryExists(extensionPathExpanded))
```